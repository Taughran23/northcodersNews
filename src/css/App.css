.header{
    border-width: 20%;
}