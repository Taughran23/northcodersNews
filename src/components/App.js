import React from 'react';
import { Router, Route } from 'react-router-dom';
import createBrowserHistory from 'history/createBrowserHistory';

import Header from './Header';
import ArticleList from './ArticleList';
import TopicsHeader from './TopicsHeader';
import TopicPage from './TopicPage';
import UserProfile from './UserProfile';
import Article from './Article';
import MobileMenu from './MobileMenu';
import Footer from './Footer';

import '../css/App.css';

const history = createBrowserHistory();

class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      active : false
    };
  }
  render () {
    return (
      <Router history={history}>
        <div>
          <Header/>
          <TopicsHeader/>
          <Route exact path='/' component={ArticleList} />
          <Route exact path='/:topic_slug' component={TopicPage} />
          <Route exact path='/users/:username' component={UserProfile} />
          <Route exact path='/articles/:article_id' component={Article} />
          <Route exact path='/menu' component={MobileMenu} />
          <Footer />
        </div>
      </Router>
    );
  }
}

export default App;