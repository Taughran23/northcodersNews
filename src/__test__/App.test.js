import React from 'react';
import Enzyme , {shallow} from 'enzyme';
import App from '../components/App';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import expect from 'expect';

Enzyme.configure({ adapter: new Adapter() });

describe('<App />', () => {
  it('should render correctly', () => {
    const component = shallow(<App />);
    expect(shallowToJson(component)).toMatchSnapshot();
  });
  it('it renders 1 <App /> component', () => {
    const component = shallow(<App />);
    expect(component).toHaveLength(1);
  });
});