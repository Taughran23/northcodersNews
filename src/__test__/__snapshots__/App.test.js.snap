// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should render correctly 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "blank",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <div>
    <Connect(TopicsHeader) />
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/:topic_slug"
    />
  </div>
</Router>
`;
